pkgs/build-support/setup-hooks/patch-ant-buildfiles.sh

```diff
@@ -0,0 +1,30 @@
+
+
+# Sets the `modificationtime` attribute on the <jar> and <war> ant tasks
+# This is done to make the final archives deterministic
+patchAntBuildfile() {
+    local buildfile="$1"
+    # using `local-name()` ignores the `xmlns` attribute in a parent node
+    @xmlstarlet@ ed --inplace \
+        --var T '//*[local-name()="jar" or local-name()="war"]' \
+        --update '$T/@modificationtime' -v 0 \
+        --insert '$T[not(@modificationtime)]' -t attr -n 'modificationtime' -v 0 \
```

Comment:
How about setting `modificationtime` to `$SOURCE_DATE_EPOCH`?

And further, if `$SOURCE_DATE_EPOCH` is not set, maybe the hook should not do anything, other than say something like "patchAntBuildfilesHook: $SOURCE_DATE_EPOCH is not set, skipping"?

Reply:
A bit unrelated to the suggestion, but what would you say, should the script only set `modificationtime` if it's unset or should it also override it if the attribute already exists? The current impl overrides, but maybe that's not needed.

Reply:
What's funny about `SOURCE_DATE_EPOCH` is that `ant` partially respects it in the [`tstamp` task](https://github.com/search?q=repo%3Aapache%2Fant%20SOURCE_DATE_EPOCH&type=code), but doesn't support it in the archive tasks. I might open an issue on upstream, and then maybe they can handle this whole ordeal, which is much less error-prone. (Though it will probably take a while before it's done)